Src/ILGPU.Algorithms.Tests/MatrixTests.cs

```diff
@@ -0,0 +1,258 @@
+// ---------------------------------------------------------------------------------------
+//                                   ILGPU Algorithms
+//                           Copyright (c) 2023 ILGPU Project
+//                                    www.ilgpu.net
+//
+// File: MatrixTests.cs
+//
+// This file is part of ILGPU and is distributed under the University of Illinois Open
+// Source License. See LICENSE.txt for details.
+// ---------------------------------------------------------------------------------------
+
+using ILGPU.Algorithms.MatrixOperations;
+using ILGPU.Runtime;
+using ILGPU.Tests;
+using System;
+using Xunit;
+using Xunit.Abstractions;
+
+#pragma warning disable CA1814 // Jagged arrays
+#pragma warning disable CA5394 // Insecure RNG
+
+namespace ILGPU.Algorithms.Tests
+{
+    public abstract partial class MatrixTests : TestBase
+    {
+        protected MatrixTests(ITestOutputHelper output, TestContext testContext)
+            : base(output, testContext)
+        { }
+
+        #region MemberData
+
+        public static TheoryData<object, object, object, object> DimensionsData =>
+            new TheoryData<object, object, object, object>
+            {
+                { 39, 17, 2918291, 0.8f },
+                { 39, 17, 2918291, 0.5f },
+                { 39, 17, 2918291, 0.05f },
+                
+                { 376, 288, 4132107, 0.8f },
+                { 376, 288, 4132107, 0.5f },
+                { 376, 288, 4132107, 0.05f },
+                
+                { 1829, 928, 31821912, 0.8f },
+                { 1829, 928, 31821912, 0.5f },
+                { 1829, 928, 31821912, 0.05f },
```

Comment:
Maybe note that this last set of tests is skipped for now?

Reply:
These tests will be skipped on the CPU but will run in the cloud on our GPUs.